cli/cmd/publicdomain/add.go

```diff
@@ -0,0 +1,66 @@
+package publicdomain
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/norman/pkg/kv"
+	"github.com/rancher/rio/cli/pkg/clicontext"
+	"github.com/rancher/rio/types/client/space/v1beta1"
+)
+
+type Add struct {
+}
+
+func (a *Add) Run(ctx *clicontext.CLIContext) error {
+	if len(ctx.CLI.Args()) != 3 {
+		return errors.New("Incorrect Usage. Example: `rio domain add NAME DOMAIN_NAME TARGET_SVC`")
```

Comment:
NAME seems redundant.  Can we just auto generate name and not show it or require it to be set?

Reply:
:+1: